documentation/uniform-versioning.md

```diff
@@ -0,0 +1,95 @@
+# Uniform versioning
+
+> [!NOTE]
+> This article uses terminology defined in the [glossary], like `service` or `service group`.
+
+A `service` **must version uniformly**. 
+
+In brief, it means:
+
+> The service, its set of operation endpoints (typically HTTP REST API endpoints), its documentation, and any SDK referencing it, all must version in lockstep.
+> If a new service API version is released, also a new documentation reference must be released to describe it.
+> Similarly, a new version of given SDK package must be released to refer to the new service version.
+
+Each `service` within a `service group` can version independently of each other.
+
+## Uniform versioning rules
+
+All of the rules listed here apply both for OpenAPI specs emitted from TypeSpec as well as for hand-written OpenAPI specs.
+
+**Uniform versioning** prescribes the following
+
+1. Any deployed service operation endpoint must belong to an API version. An API version once deployed is immutable.
```

Comment:
@konrad-jamrozik I have two questions regarding this:
- Considering we have no issues in service side but issues in defined spec in one version, can we directly update spec to fix this spec issue?
- Considering we have issues in service side and relevant spec in one version, can we directly fix the issue in service side and update the spec in the same api version?

Reply:
@MaryGao:

Re

> Imagine we have no issues in service side but issues in defined spec in v1, can we directly update spec to fix this spec issue in v1?

Yes, you can. If there is a breaking change caused by the spec update to match the service behavior, you can approve it with `BreakingChange-Approved-BugFix` or `Versioning-Approved-BugFix`. Details:

- https://aka.ms/azsdk/pr-brch-deep#breakingchange-approved-bugfix (`BreakingChange-Approved-BugFix`)
- https://aka.ms/azsdk/pr-brch-deep#versioning-approved-bugfix (`Versioning-Approved-BugFix`)

Re 

> Imagine we have issues in service side and relevant spec in v2, can we directly fix the issue in service side and update the spec in the same api version v2?

I believe the answer is: no, you can not. But if this was a `preview` API version, you can introduce a new API version without going through the formal "breaking change notification" process (the green "ignore" box in [this diagram](https://eng.ms/docs/products/azure-developer-experience/design/specs-pr-guides/pr-brch-deep#diagram-explaining-breaking-changes-and-versioning-issues)). However, I am not 100% sure about this guidance - depending of the exact nature of the bug the Breaking Change Board may possibly allow you to do it. Let me defer to @mikekistler and @JeffreyRichter to clarify the guidance here.

Reply:
On the second point, if V2 is deployed and available to customers then its behavior should not change. If the behavior is "wrong" and needs to be fixed that should be done in a new api-version.